cluster-operator/src/main/java/io/strimzi/operator/cluster/model/CruiseControl.java

```diff
@@ -165,8 +165,8 @@ public class CruiseControl extends AbstractModel {
      * @param resource  Kubernetes resource with metadata containing the namespace and cluster name
      */
     protected CruiseControl(Reconciliation reconciliation, HasMetadata resource) {
-        super(reconciliation, resource, APPLICATION_NAME);
-        this.name = CruiseControlResources.deploymentName(cluster);
+        super(reconciliation, resource, CruiseControlResources.deploymentName(resource.getMetadata().getName()), APPLICATION_NAME);
```

Comment:
should we change the `APPLICATION_NAME` constant to `COMPONENT_TYPE` now? or vice versa, kind of related to my comment on Javadoc before?